python/graphscope/framework/context.py

```diff
@@ -190,21 +165,93 @@ def to_vineyard_dataframe(self, selector=None, vertex_range=None):
         Returns:
             str: object id of vineyard tensor
         """
-        self._check_unmodified()
         if selector is not None:
-            check_argument(
-                isinstance(selector, Mapping),
-                "selector of to_vineyard_dataframe must be a dict",
-            )
-            selector = {
-                key: self._transform_selector(value) for key, value in selector.items()
-            }
             selector = json.dumps(selector)
         vertex_range = utils.transform_vertex_range(vertex_range)
         op = dag_utils.to_vineyard_dataframe(self, selector, vertex_range)
-        ret = op.eval()
-        object_id = json.loads(ret)["object_id"]
-        return object_id
+        return ResultDAGNode(self, op)
+
+
+class Context(collections.abc.Mapping):
+    def __init__(self, context_node, key, type):
+        self._context_node = context_node
+        self._session = context_node.session
+        self._session_id = self._session.session_id
+        self._graph = self._context_node._graph
+        self._key = key
+        self._type = type
+        # copy and set op evaluated
+        self._context_node.op = deepcopy(self._context_node.op)
+        self._context_node.evaluated = True
+        self._saved_signature = self.signature
+
+    @property
+    def session_id(self):
+        return self._session_id
+
+    @property
+    def op(self):
+        return self._context_node.op
+
+    @property
+    def key(self):
+        """Unique identifier of a context."""
+        return self._key
+
+    @property
+    def signature(self):
+        """Compute digest by key and graph signatures.
+        Used to ensure the critical information of context is untouched.
+        """
+        check_argument(
+            self._key is not None,
+            "Context key error, maybe it is not connected to engine.",
+        )
+        return hashlib.sha256(
+            "{}.{}".format(self._key, self._graph.signature).encode("utf-8")
+        ).hexdigest()
+
+    def __repr__(self):
+        return f"graphscope.{self.__class__.__name__} from graph {str(self._graph)}"
+
+    def __len__(self):
+        return self._graph._graph.number_of_nodes()
+
+    def __getitem__(self, key):
```

Comment:
the __getitem__ actually only support in dynamicContext, not for other Context type.

Reply:
Distinguish them in different class